### ISSUE TITLE
How to use (and preferably finetune) BART for text infilling? 

### ISSUE THREAD

**Issue by tomaszgarbus** [huggingface/transformers#5428]
[Here](https://huggingface.co/transformers/model_doc/bart.html#bartforconditionalgeneration) is shown how to use BART for simple mask filling (one <mask> token = one generated token), but how to use it for text infilling? The BART paper states that the model was pretrained on such task so it should be possible.

Is the only solution to simply take the `facebook/bart-large` model for summarization and finetune it on a dataset with <mask> tokens or is there a better way?

**Comment by orena1** [comment]
@julien-c , @sshleifer ?

**Comment by sshleifer** [comment]
Unfortunately, text infilling is not yet supported. It would be a welcome contribution! I think the equivalent fairseq task is called `DenoisingTask` 
https://github.com/pytorch/fairseq/blob/aa79bb9c37b27e3f84e7a4e182175d3b50a79041/fairseq/tasks/denoising.py#L27
